src/vivarium/framework/lifecycle.py

```diff
@@ -376,7 +376,7 @@ def to_guid(method: MethodType) -> str:
         has been constrained before.
 
         """
-        return f'{str(id(method.__self__))}.{method.__name__}'
+        return f'{method.__self__.name}.{method.__name__}'
```

Comment:
This is the constraint maker fix.  If something like a population view gets garbage collected, it's id gets repurposed by python.